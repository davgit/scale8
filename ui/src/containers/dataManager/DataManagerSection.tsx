import { FC } from 'react';
import { useQuery } from '@apollo/client';
import {
    BreadcrumbButtonProps,
    buildDataManagerButtonProps,
} from '../../utils/BreadcrumbButtonsUtils';
import NavDataManagerQuery from '../../gql/queries/NavDataManagerQuery';
import { NavDataManager } from '../../gql/generated/NavDataManager';
import { Section, SectionItem, SectionProps } from '../abstractions/Section';
import { SectionKey } from '../SectionsDetails';
import { useConfigState, useLoggedInState } from '../../context/AppContext';
import { NextRouter, useRouter } from 'next/router';
import { ChildrenAndIdProps } from '../../types/props/ChildrenAndIdProps';
import { CurrentOrgPermissions } from '../../context/OrgUserReducer';
import { buildOrgButtons } from '../global/OrgSection';

export const buildDataManagerButtons = (
    orgs: SectionItem[],
    currentOrg: SectionItem,
    tagManagerId: string,
    dataManagerId: string,
    router: NextRouter,
    orgPermissions: CurrentOrgPermissions,
    useSignup: boolean,
    isCurrentPage = false,
): BreadcrumbButtonProps[] => [
    ...buildOrgButtons(orgs, currentOrg, router, orgPermissions, useSignup, 'Services'),
    buildDataManagerButtonProps(router, dataManagerId, tagManagerId, isCurrentPage),
];

const DataManagerSection: FC<ChildrenAndIdProps> = (props: ChildrenAndIdProps) => {
    const router = useRouter();
    const { id, children } = props;

    const { orgUserState } = useLoggedInState();
    const { useSignup } = useConfigState();

    const sectionProps: SectionProps<NavDataManager> = {
        children,
        sectionKey: SectionKey.dataManager,
        queryResult: useQuery<NavDataManager>(NavDataManagerQuery, {
            variables: { id },
        }),
        buildButtonsProps: (data, orgPermissions) => {
            return buildDataManagerButtons(
                data.me.orgs,
                data.getDataManagerAccount.org,
                data.getDataManagerAccount.org.tag_manager_account?.id ?? '',
                id,
                router,
                orgPermissions,
                useSignup,
                true,
            );
        },
        buildMenuItemsProps: () => [],
        extractOrgUserDetails: (data) => data.getDataManagerAccount.org,
        accountExpireIn: orgUserState?.dataManagerAccount?.trialExpiration ?? undefined,
        accountIsTrial: orgUserState?.dataManagerAccount?.isTrial ?? undefined,
    };

    return <Section<NavDataManager> {...sectionProps} />;
};

export default DataManagerSection;
