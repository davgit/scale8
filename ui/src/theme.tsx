import { createTheme, adaptV4Theme } from '@mui/material/styles';
import { getProductSection, ProductSectionKey } from './containers/SectionsDetails';
import { grey } from '@mui/material/colors';

declare module '@mui/material/styles/createPalette' {
    interface Palette {
        adminColor: Palette['primary'];
        dataManagerColor: Palette['primary'];
        tagManagerColor: Palette['primary'];
        commonColor: Palette['primary'];
    }
    interface PaletteOptions {
        adminColor: PaletteOptions['primary'];
        dataManagerColor: PaletteOptions['primary'];
        tagManagerColor: PaletteOptions['primary'];
        commonColor: PaletteOptions['primary'];
    }
}

// A custom theme for this app
const theme = createTheme(
    adaptV4Theme({
        palette: {
            primary: {
                main: '#25302e',
            },
            secondary: {
                main: grey['700'],
            },
            error: {
                main: '#c63d51',
            },
            info: {
                main: '#444444',
            },
            success: {
                main: '#c6e54b',
            },
            background: {
                default: '#ffffff',
            },
            adminColor: {
                main: getProductSection(ProductSectionKey.admin).color,
            },
            dataManagerColor: {
                main: getProductSection(ProductSectionKey.dataManager).color,
            },
            tagManagerColor: {
                main: getProductSection(ProductSectionKey.tagManager).color,
            },
            commonColor: {
                main: getProductSection(ProductSectionKey.global).color,
            },
        },
        typography: {
            fontFamily: ['Source Sans Pro', 'Helvetica Neue', 'Arial', 'sans-serif'].join(','),
            fontSize: 14,
            body1: {
                fontSize: 14,
            },
        },
        overrides: {
            MuiTableCell: {
                head: {
                    fontWeight: 600,
                },
            },
            MuiSvgIcon: {
                root: {
                    fontSize: '1.25rem',
                },
            },
            MuiButton: {
                root: {
                    fontWeight: 600,
                },
            },
            MuiTooltip: {
                tooltip: {
                    fontSize: '.8rem',
                },
            },
            MuiDialogActions: {
                spacing: {
                    '& > :not(:first-child)': {
                        marginLeft: '16px',
                    },
                },
            },
        },
    }),
);

export default theme;
