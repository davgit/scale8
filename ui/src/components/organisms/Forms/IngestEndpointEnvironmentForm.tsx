import { FC } from 'react';
import ControlledTextInput from '../../atoms/ControlledInputs/ControlledTextInput';
import ControlledSelect from '../../atoms/ControlledInputs/ControlledSelect';
import DrawerFormLayout from '../../molecules/DrawerFormLayout';
import { IngestEndpointEnvironmentFormProps } from '../../../dialogPages/dataManager/IngestEndpointEnvironmentCreate';
import StorageProviderSelector from '../../molecules/StorageProviderSelector';

const IngestEndpointEnvironmentForm: FC<IngestEndpointEnvironmentFormProps> = (
    props: IngestEndpointEnvironmentFormProps,
) => {
    return (
        <DrawerFormLayout {...props} submitDisable={props.availableRevisions.length < 1}>
            <ControlledTextInput
                name="name"
                label="Name"
                formProps={props}
                className="DrawerFormField"
                required
                autoFocus
            />

            {props.availableRevisions.length < 1 && (
                <small>
                    At least one revision needs to be finalised for it to be attached to an
                    environment.
                </small>
            )}

            <ControlledSelect
                className="DrawerFormField"
                label="Revision"
                name="revisionId"
                values={props.availableRevisions}
                formProps={props}
                disabled={props.availableRevisions.length < 1}
                required
            />

            <StorageProviderSelector {...props} includeBigQueryPartitionFilter />
        </DrawerFormLayout>
    );
};

export default IngestEndpointEnvironmentForm;
