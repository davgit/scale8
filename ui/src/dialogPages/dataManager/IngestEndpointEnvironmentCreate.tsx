import { Dispatch, FC, SetStateAction } from 'react';
import IngestEndpointEnvironmentForm from '../../components/organisms/Forms/IngestEndpointEnvironmentForm';
import { FormProps, FormValidationResult } from '../../hooks/form/useFormValidation';
import { ApolloError, useMutation, useQuery } from '@apollo/client';
import CreateIngestEndpointEnvironmentQuery from '../../gql/mutations/CreateIngestEndpointEnvironmentQuery';
import FetchIngestEndpointRevisionsQuery from '../../gql/queries/FetchIngestEndpointRevisionsQuery';
import { IngestEndpointRevisionsData } from '../../gql/generated/IngestEndpointRevisionsData';
import {
    IngestEndpointEnvironmentCreateInput,
    Mode,
    StorageProvider,
} from '../../gql/generated/globalTypes';
import nameValidator from '../../utils/validators/nameValidator';
import { CreateIngestEndpointEnvironmentResult } from '../../gql/generated/CreateIngestEndpointEnvironmentResult';
import domainValidator from '../../utils/validators/domainValidator';
import { buildStandardFormInfo } from '../../utils/InfoLabelsUtils';
import { DialogPageProps } from '../../types/DialogTypes';
import { useConfigState } from '../../context/AppContext';
import { DialogPreloadForm, DialogPreloadFormProps } from '../abstractions/DialogPreloadForm';
import {
    buildStorageProviderSaveProperties,
    initialStorageProviderFieldsWithPartitionFilterChoice,
    storageProviderCustomValueSetter,
    StorageProviderFieldsWithPartitionFilterChoice,
    storageProviderValidators,
} from '../../utils/StorageProviderUtils';

export type IngestEndpointEnvironmentValues = StorageProviderFieldsWithPartitionFilterChoice & {
    name: string;
    revisionId: string;
    domain?: string;
    certificate: string;
    key: string;
};

export type IngestEndpointEnvironmentFormProps = FormProps<IngestEndpointEnvironmentValues> & {
    isCreate: boolean;
    availableRevisions: { key: string; text: string }[];
    hasCustomDomain?: boolean;
};

const customValueSetter = (
    valueKey: string,
    value: any,
    values: IngestEndpointEnvironmentValues,
    setValues: Dispatch<SetStateAction<IngestEndpointEnvironmentValues>>,
) => {
    if (storageProviderCustomValueSetter(valueKey, value, values, setValues)) {
        return;
    }
    if (valueKey === 'domain') {
        setValues({
            ...values,
            [valueKey]: value,
            certificate: value === '' ? '' : values.certificate,
            key: value === '' ? '' : values.key,
        });
    } else {
        setValues({
            ...values,
            [valueKey]: value,
        });
    }
};

const IngestEndpointEnvironmentCreate: FC<DialogPageProps> = (props: DialogPageProps) => {
    const ingestEndpointId = props.contextId;
    const { mode } = useConfigState();

    const ingestEndpointEnvironmentCreateProps: DialogPreloadFormProps<
        IngestEndpointRevisionsData,
        IngestEndpointEnvironmentValues,
        IngestEndpointEnvironmentFormProps,
        CreateIngestEndpointEnvironmentResult
    > = {
        loadQuery: useQuery<IngestEndpointRevisionsData>(FetchIngestEndpointRevisionsQuery, {
            variables: { id: ingestEndpointId },
        }),
        buildInitialStatePreload: () => ({
            ...initialStorageProviderFieldsWithPartitionFilterChoice,
            name: '',
            revisionId: '',
            domain: '',
            certificate: '',
            key: '',
        }),
        saveQuery: useMutation(CreateIngestEndpointEnvironmentQuery),
        mapSaveData: (ingestEndpointEnvironmentValues: IngestEndpointEnvironmentValues) => {
            const ingestEndpointEnvironmentCreateInput: IngestEndpointEnvironmentCreateInput = {
                ingest_endpoint_id: ingestEndpointId,
                ingest_endpoint_revision_id: ingestEndpointEnvironmentValues.revisionId,
                storage_provider:
                    ingestEndpointEnvironmentValues.storageProvider as StorageProvider,
                name: ingestEndpointEnvironmentValues.name,
                ...buildStorageProviderSaveProperties(ingestEndpointEnvironmentValues, true, true),
            };

            if (ingestEndpointEnvironmentValues.domain !== '') {
                ingestEndpointEnvironmentCreateInput.custom_domain =
                    ingestEndpointEnvironmentValues.domain;
                ingestEndpointEnvironmentCreateInput.cert_pem =
                    ingestEndpointEnvironmentValues.certificate;
                ingestEndpointEnvironmentCreateInput.key_pem = ingestEndpointEnvironmentValues.key;
            }

            return { ingestEndpointEnvironmentCreateInput };
        },
        buildFormProps: (
            formLoadedData: IngestEndpointRevisionsData,
            formValidationValues: FormValidationResult<IngestEndpointEnvironmentValues>,
            gqlError?: ApolloError,
        ) => {
            const filteredRevisions =
                formLoadedData.getIngestEndpoint.ingest_endpoint_revisions.filter((_) => _.locked);

            return {
                ...formValidationValues,
                gqlError,
                submitText: 'Create Environment',
                title: 'Create Environment',
                availableRevisions: filteredRevisions.map((_) => ({
                    key: _.id,
                    text: _.name,
                })),
                isCreate: true,
                formInfoProps: buildStandardFormInfo('ingestEndpointEnvironments', 'Create'),
                handleDialogClose: props.handleDialogClose,
                hasCustomDomain: mode === Mode.COMMERCIAL ? undefined : false,
            };
        },
        checkSuccessfullySubmitted: (formMutationData) =>
            formMutationData?.createIngestEndpointEnvironment.id !== undefined,
        pageComponent: IngestEndpointEnvironmentForm,
        validators: [
            {
                field: 'name',
                validator: nameValidator,
                error: () => 'Ingest Endpoint Environment name too short',
            },
            {
                field: 'domain',
                validator: domainValidator,
                error: () => 'Invalid domain',
                ignoreEmpty: true,
            },
            ...storageProviderValidators,
        ],
        customValueSetter,
        ...props,
    };

    return (
        <DialogPreloadForm<
            IngestEndpointRevisionsData,
            IngestEndpointEnvironmentValues,
            IngestEndpointEnvironmentFormProps,
            CreateIngestEndpointEnvironmentResult
        >
            {...ingestEndpointEnvironmentCreateProps}
        />
    );
};

export { IngestEndpointEnvironmentCreate };
