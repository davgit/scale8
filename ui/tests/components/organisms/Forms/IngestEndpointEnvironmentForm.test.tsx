import { render } from '@testing-library/react';
import { ThemeProvider } from '@mui/material';
import theme from '../../../../src/theme';
import { FC } from 'react';
import { useMockFormProps } from '../../../componentsMocks/useMockFormProps';
import IngestEndpointEnvironmentForm from '../../../../src/components/organisms/Forms/IngestEndpointEnvironmentForm';
import { IngestEndpointEnvironmentValues } from '../../../../src/dialogPages/dataManager/IngestEndpointEnvironmentCreate';

const MockForm: FC = () => {
    const formProps = useMockFormProps<IngestEndpointEnvironmentValues>({
        name: '',
        revisionId: '',
        certificate: '',
        key: '',
        storageProvider: 'AWS_KINESIS',
        region: '',
    });

    return (
        <IngestEndpointEnvironmentForm
            {...formProps}
            isCreate={true}
            availableRevisions={[{ key: 'p1', text: 'p1' }]}
        />
    );
};

describe('IngestEndpointEnvironmentForm', () => {
    afterEach(() => {
        jest.clearAllMocks();
    });
    it('form render unchanged', () => {
        const { asFragment } = render(
            <ThemeProvider theme={theme}>
                <MockForm />
            </ThemeProvider>,
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
