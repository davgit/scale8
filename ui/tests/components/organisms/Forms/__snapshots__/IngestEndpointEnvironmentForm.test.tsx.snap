// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IngestEndpointEnvironmentForm form render unchanged 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-s92abg"
  >
    <div
      class="MuiBox-root css-70qvj9"
    >
      <div
        class="MuiBox-root css-1isemmb"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-erxxaa-MuiSvgIcon-root"
            data-testid="CloseIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiBox-root css-uix0gv"
      >
        mock form
      </div>
    </div>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
    />
    <div
      class="MuiBox-root css-yn9cjk"
    >
      <div
        class="MuiBox-root css-1676va5"
      >
        <form
          class="MuiBox-root css-mjj15b"
        >
          <div
            class="MuiFormControl-root MuiTextField-root DialogFormField css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary Mui-focused Mui-required css-1oyovms-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              for="mui-1"
              id="mui-1-label"
            >
              Name
              <span
                aria-hidden="true"
                class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-b4y4zd-MuiFormLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary Mui-focused MuiInputBase-formControl MuiInputBase-adornedEnd css-1v2pam7-MuiInputBase-root-MuiInput-root"
            >
              <input
                aria-invalid="false"
                autocomplete="off"
                class="MuiInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-1x51dt5-MuiInputBase-input-MuiInput-input"
                id="mui-1"
                name="name"
                required=""
                type="text"
                value=""
              />
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
              >
                <button
                  aria-label="toggle password visibility"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-erxxaa-MuiSvgIcon-root"
                    data-testid="ClearIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
          <div
            class="MuiFormControl-root DialogFormField css-1nrlq1o-MuiFormControl-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-a6y1oi-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
            >
              Revision
              <span
                aria-hidden="true"
                class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-b4y4zd-MuiFormLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-foxv06-MuiInputBase-root-MuiInput-root-MuiSelect-root"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="mui-component-select-revisionId"
                class="MuiSelect-select MuiSelect-standard MuiInput-input MuiInputBase-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
                id="mui-component-select-revisionId"
                role="button"
                tabindex="0"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                name="revisionId"
                required=""
                tabindex="-1"
                value=""
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-yqkyv6-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
          <div
            class="MuiFormControl-root DialogFormField css-1nrlq1o-MuiFormControl-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-required css-1oyovms-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
            >
              Storage Provider
              <span
                aria-hidden="true"
                class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-b4y4zd-MuiFormLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-foxv06-MuiInputBase-root-MuiInput-root-MuiSelect-root"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="mui-component-select-storageProvider"
                class="MuiSelect-select MuiSelect-standard MuiInput-input MuiInputBase-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
                id="mui-component-select-storageProvider"
                role="button"
                tabindex="0"
              >
                Amazon Kinesis
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                name="storageProvider"
                required=""
                tabindex="-1"
                value="AWS_KINESIS"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-yqkyv6-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
          <div
            class="MuiBox-root css-17ck9e8"
          >
            <div
              class="MuiBox-root css-79elbk"
            >
              <div
                class="MuiBox-root css-rxw2ex"
              >
                AWS Kinesis Storage Config
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root DialogFormField css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-a6y1oi-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="mui-2"
                id="mui-2-label"
              >
                Stream Name
                <span
                  aria-hidden="true"
                  class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-b4y4zd-MuiFormLabel-asterisk"
                >
                   *
                </span>
              </label>
              <div
                class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-1v2pam7-MuiInputBase-root-MuiInput-root"
              >
                <input
                  aria-invalid="false"
                  autocomplete="off"
                  class="MuiInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-1x51dt5-MuiInputBase-input-MuiInput-input"
                  id="mui-2"
                  name="streamName"
                  required=""
                  type="text"
                  value=""
                />
                <div
                  class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
                >
                  <button
                    aria-label="toggle password visibility"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-erxxaa-MuiSvgIcon-root"
                      data-testid="ClearIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root DialogFormField css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-a6y1oi-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="mui-3"
                id="mui-3-label"
              >
                Access Key Id
                <span
                  aria-hidden="true"
                  class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-b4y4zd-MuiFormLabel-asterisk"
                >
                   *
                </span>
              </label>
              <div
                class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-1v2pam7-MuiInputBase-root-MuiInput-root"
              >
                <input
                  aria-invalid="false"
                  autocomplete="off"
                  class="MuiInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-1x51dt5-MuiInputBase-input-MuiInput-input"
                  id="mui-3"
                  name="accessKeyId"
                  required=""
                  type="text"
                  value=""
                />
                <div
                  class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
                >
                  <button
                    aria-label="toggle password visibility"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-erxxaa-MuiSvgIcon-root"
                      data-testid="ClearIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root DialogFormField css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-a6y1oi-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="mui-4"
                id="mui-4-label"
              >
                Secret Access Key
                <span
                  aria-hidden="true"
                  class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-b4y4zd-MuiFormLabel-asterisk"
                >
                   *
                </span>
              </label>
              <div
                class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-1v2pam7-MuiInputBase-root-MuiInput-root"
              >
                <input
                  aria-invalid="false"
                  autocomplete="off"
                  class="MuiInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-1x51dt5-MuiInputBase-input-MuiInput-input"
                  id="mui-4"
                  name="secretAccessKey"
                  required=""
                  type="text"
                  value=""
                />
                <div
                  class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
                >
                  <button
                    aria-label="toggle password visibility"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-erxxaa-MuiSvgIcon-root"
                      data-testid="ClearIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </div>
            <div
              class="MuiFormControl-root DialogFormField css-1nrlq1o-MuiFormControl-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-a6y1oi-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
              >
                Region
                <span
                  aria-hidden="true"
                  class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-b4y4zd-MuiFormLabel-asterisk"
                >
                   *
                </span>
              </label>
              <div
                class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-foxv06-MuiInputBase-root-MuiInput-root-MuiSelect-root"
              >
                <div
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  aria-labelledby="mui-component-select-region"
                  class="MuiSelect-select MuiSelect-standard MuiInput-input MuiInputBase-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
                  id="mui-component-select-region"
                  role="button"
                  tabindex="0"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </div>
                <input
                  aria-hidden="true"
                  class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                  name="region"
                  required=""
                  tabindex="-1"
                  value=""
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-yqkyv6-MuiSvgIcon-root-MuiSelect-icon"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="MuiBox-root css-wpygli"
          >
            <button
              class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-xcbv7o-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="submit"
            >
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
